src/cli/crd/generate.ts

```diff
@@ -0,0 +1,316 @@
+    for (const sourceFile of sourceFiles) {
+      processSourceFile(sourceFile, version, outputDir);
+    }
+  }
+}
+
+export function getAPIVersions(apiRoot: string): string[] {
+  return fs.readdirSync(apiRoot).filter(v => fs.statSync(path.join(apiRoot, v)).isDirectory());
+}
+
+export function loadVersionFiles(project: Project, versionDir: string): SourceFile[] {
+  const files = fs.readdirSync(versionDir).filter(f => f.endsWith(".ts"));
+  const filePaths = files.map(f => path.join(versionDir, f));
+  return project.addSourceFilesAtPaths(filePaths);
+}
+
+export function processSourceFile(
+  sourceFile: SourceFile,
+  version: string,
+  outputDir: string,
+): void {
+  const content = sourceFile.getFullText();
+  const { kind, fqdn, scope, plural, shortNames } = extractCRDDetails(content, sourceFile);
+
+  if (!kind) {
+    console.warn(`⚠️ Skipping ${sourceFile.getBaseName()}: missing '// Kind: <KindName>' comment`);
+    return;
+  }
+
+  const spec = sourceFile.getInterface(`${kind}Spec`);
+  if (!spec) {
+    console.warn(`⚠️ Skipping ${sourceFile.getBaseName()}: missing interface ${kind}Spec`);
+    return;
+  }
+
+  const condition = sourceFile.getTypeAlias(`${kind}StatusCondition`);
+  const specSchema = getSchemaFromType(spec);
+  const conditionSchema = condition ? getSchemaFromType(condition) : emptySchema();
+
+  const crd = buildCRD({
+    kind,
+    fqdn,
+    version,
+    plural,
+    scope,
+    shortNames,
+    specSchema,
+    conditionSchema,
+  });
+
+  const outPath = path.join(outputDir, `${kind.toLowerCase()}.yaml`);
+  fs.writeFileSync(outPath, toYAML(crd), "utf8");
+  console.log(`✔ Created ${outPath}`);
+}
+
+// Extracts a comment from the content of a file based on a label.
+export function extractSingleLineComment(content: string, label: string): string | undefined {
+  // https://regex101.com/r/oLFaHP/1
+  const match = content.match(new RegExp(`//\\s+${label}:\\s+(.*)`));
+  return match?.[1].trim();
+}
+
+export function extractDetails(sourceFile: SourceFile): {
+  plural: string;
+  scope: "Cluster" | "Namespaced";
+  shortName: string;
+} {
+  const decl = sourceFile.getVariableDeclaration("details");
+  if (!decl) {
+    throw new Error(`Missing 'details' variable declaration.`);
+  }
+
+  const init = decl.getInitializerIfKindOrThrow(SyntaxKind.ObjectLiteralExpression);
+
+  const getStr = (key: string): string => {
+    const prop = init.getProperty(key);
+    const value = prop?.getFirstChildByKind(SyntaxKind.StringLiteral)?.getLiteralText();
+    if (!value) {
+      throw new Error(`Missing or invalid value for required key: '${key}'`);
+    }
+    return value;
+  };
+
+  const scope = getStr("scope");
+  if (scope !== "Cluster" && scope !== "Namespaced") {
+    throw new Error(`'scope' must be either "Cluster" or "Namespaced", got "${scope}"`);
+  }
+
+  return {
+    plural: getStr("plural"),
+    scope,
+    shortName: getStr("shortName"),
+  };
```

Comment:
```suggestion
  if (scope === "Cluster" || scope === "Namespaced") {
    return {
      plural: getStr("plural"),
      scope,
      shortName: getStr("shortName"),
    };
  }

  throw new Error(`'scope' must be either "Cluster" or "Namespaced", got "${scope}"`);
```

This proposal rewrites the conditional in the positive case instead of the negative case.

Reply:
i personally don't care if the error or the return gets returned first but i guess so!